grid-proxy/tools/db/generate.go

```diff
@@ -216,47 +324,74 @@ func generateNameContracts(db *sql.DB) error {
 				state = "GracePeriod"
 			}
 		}
-		twinID := rnd(1100, 3100)
+
+		twinID, err := rnd(1100, 3100)
+		if err != nil {
+			return err, billsStartID, contractsStartID
+		}
+
 		if renter, ok := renter[nodeID]; ok {
 			twinID = renter
 		}
+
 		if _, ok := availableRentNodes[nodeID]; ok {
 			i--
 			continue
 		}
+
 		contract := name_contract{
-			id:           fmt.Sprintf("node-contract-%d", contractCnt),
+			id:           fmt.Sprintf("name-contract-%d", contractsStartID),
 			twin_id:      twinID,
-			contract_id:  contractCnt,
+			contract_id:  uint64(contractsStartID),
 			state:        state,
 			created_at:   uint64(time.Now().Unix()),
 			grid_version: 3,
 			name:         uuid.NewString(),
 		}
-		if _, err := db.Exec(insertQuery(&contract)); err != nil {
-			panic(err)
+
+		contracts = append(contracts, objectToTupleString(contract))
+
+		billings, err := rnd(0, 10)
+		if err != nil {
+			return err, billsStartID, contractsStartID
 		}
-		billings := rnd(0, 10)
+		amountBilled, err := rnd(0, 100000)
 		for j := uint64(0); j < billings; j++ {
 			billing := contract_bill_report{
-				id:                fmt.Sprintf("contract-bill-report-%d", billCnt),
-				contract_id:       contractCnt,
+				id:                fmt.Sprintf("contract-bill-report-%d", billsStartID),
+				contract_id:       uint64(contractsStartID),
 				discount_received: "Default",
-				amount_billed:     rnd(0, 100000),
+				amount_billed:     amountBilled,
 				timestamp:         uint64(time.Now().UnixNano()),
 			}
-			billCnt++
-			if _, err := db.Exec(insertQuery(&billing)); err != nil {
-				panic(err)
-			}
+			billsStartID++
+
+			billReports = append(billReports, objectToTupleString(billing))
 		}
-		contractCnt++
+		contractsStartID++
 	}
-	return nil
+
+	if err := insertTuples(db, name_contract{}, contracts); err != nil {
+		return err, billsStartID, contractsStartID
+	}
+
+	if err := insertTuples(db, contract_bill_report{}, billReports); err != nil {
+		return err, billsStartID, contractsStartID
+	}
+
+	fmt.Println("name contracts generated")
+
+	return nil, billsStartID, contractsStartID
 }
-func generateRentContracts(db *sql.DB) error {
+func generateRentContracts(db *sql.DB, billsStartID, contractsStartID int) (error, int, int) {
```

Comment:
nothing mandates that an error should have a specific position an a function's returned values, but if we care about our code's readability we would stick with the convention of having the error as the last returned value, right?